File: 404/404.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        queue = deque([root])
        left = 0
        
        while queue:
            for _ in range(len(queue)):
                node = queue.popleft()
                if node.left:
                    if not node.left.left and not node.left.right:
                        left+=node.left.val
                    else:
                        queue.append(node.left)
                if node.right:
                    queue.append(node.right)
        return left

File: 404/404.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function sumOfLeftLeaves(root: TreeNode | null): number {
    let queue = [root];
    let left = 0;
    while (queue.length>0){
        for (let i = 0; i < queue.length; i++){
            let node = queue.shift();
            if (node.left){
                if ((!node.left.left) && (!node.left.right)){
                    left += node.left.val
                }
                queue.push(node.left);
            }
            if (node.right){
                queue.push(node.right)}}}
    return left;};
